scripts/apps/authoring/authoring/directives/VideoTimeline.ts

```diff
@@ -0,0 +1,317 @@
+                        width: (position * 100) + '%'
+                    });
+                    barleft.setAttribute("data-content", getstrtime(position * scope.video.duration));
+                    scope.cut.start = position * scope.video.duration;
+                }
+            };
+
+
+            function onDragEndCb() {
+                setTimeline();
+                scope.onChange();
+            };
+
+            function onDragStart() {
+                var img = document.createElement("img");
+                event.dataTransfer.setDragImage(img, 0, 0);
+                event.dataTransfer.setData('text/plain', null)
+            };
+
+
+            function setTimeline() {
+                var position = getPositionBar(event.clientX);
+                scope.video.currentTime = position * scope.video.duration;
+                inner.innerHTML = getstrtime(scope.video.currentTime);
+                TweenMax.set(progressoutput, {
+                    left: (position * 100) + '%'
+                });
+                TweenMax.set(progressoutput, {
+                    left: (position * 100) + '%'
+                });
+                return position;
+            };
+
+            function getRandomSpan() {
+                let random = Math.floor(1000 + Math.random() * 9000);
+                return random
+            }
+
+            function delay(ms) {
+                return new Promise(resolve => setTimeout(resolve, ms));
+            }
+            const stopInterval = (id) => {
+                $interval.cancel(id);
+                id = undefined;
+            };
+
+            $rootScope.$on('$destroy', () => stopInterval(IntervalID))
+
+            function loadTimeLine(list_thumbnails) {
+                if (controlbar.offsetWidth <= 0) {
+                    return;
+                }
+                let widthpic = 50 * scope.video.clientWidth / scope.video.clientHeight;
+                let total_thumbnail = Math.floor(controlbar.offsetWidth / widthpic);
+                let number = getRandomSpan()
+                inner_frames.innerHTML = '';
+                let count = 0;
+                let per_delta_image;
+                let time = 0
+                if (list_thumbnails && list_thumbnails.length > 0) {
+                    per_delta_image = (list_thumbnails.length - 1) / total_thumbnail;
+                }
+                else {
+                    per_delta_image = scope.video.duration / total_thumbnail;
+                    time = 1000;
+                }
+                let thumnails = [];
+                for (let i = 0; i <= total_thumbnail; i++) {
+                    let video = document.createElement("video");
+                    video.width = widthpic;
+                    video.height = 50;
+                    thumnails.push(video)
+                    inner_frames.append(video);
+                }
+                IntervalID = $interval(async function () {
```

Comment:
When the list frames haven't created on server yet (the video's just edited), list frames will get directive position of video stream on server. So if we call get all frames at same time to server, it will make a lag and slow, sometime raise timeout and missing picture. We see it work well if we load each of them.

Reply:
> So if we call get all frames at same time to server, it will make a lag and slow, sometime raise timeout and missing picture.

That sounds like an issue that needs to be fixed on the server. Please do that and remove the interval. If you need to wait for the thumnails to be generated, send a HTTP request and render all thumbnail images when you receive a respone.